### PR TITLE
feat(Consents): Added new-style consents to the user profile page.

### PR DESCRIPTION
**Pull request by lauryndbrown**
Fixes https://github.com/carpentries/amy/issues/1925
Screen Shot 2021-05-24 at 10 10 30 AM

### CODE REVIEW

File: amy/workshops/views.py [hunk 1]
```diff
@@ -286,7 +286,22 @@ def get_context_data(self, **kwargs):
...
         context["is_usersocialauth_in_sync"] = is_usersocialauth_in_sync
+        consents = (
+            Consent.objects.filter(person=self.object)
+            .active()
+            .select_related("term", "term_option")
+        )
 
+        context["consents"] = {
+            "May contact": consents.filter(term__slug="may-contact")[0],
+            "Consent to publish profile": consents.filter(term__slug="public-profile")[
```
Comment (lauryndbrown) [inline amy/workshops/views.py:297]: Solution seems a little strange. Happy to put this in the database as an extra field instead.

Comment (pbanaszkiewicz) [reply amy/workshops/views.py:297]: Do we have a way to figure out the term question, e.g. "May contact"?

On a related note, I think using a mapping between `term.slug` and a label ("May contact") should be used to reduce the number of queries.

Comment (lauryndbrown) [reply amy/workshops/views.py:297]: Yes. I'm just not storing a label with the same wording is all. for instance the privacy policy is: *I have read and agree to the data privacy policy of The Carpentries.* I'll revise this. Maybe it's not as big of an issue if they don't map perfectly.

File: amy/workshops/views.py [hunk 2]
```diff
@@ -286,7 +286,22 @@ def get_context_data(self, **kwargs):
...
+                0
+            ],
+            "Consent to include name when publishing lessons": consents.filter(
+                term__slug="may-publish-name"
+            )[0],
+            "Privacy policy agreement": consents.filter(term__slug="privacy-policy")[0],
```
Comment (pbanaszkiewicz) [inline amy/workshops/views.py:303]: This PR looks good, except for the additional queries the `context["consents"]` introduces - can you perhaps improve to no more than 1 query?

**Comment by lauryndbrown** [comment]
@pbanaszkiewicz that should be taken care of now.

**Comment by lauryndbrown** [comment]
Also noticed that it was a little confusing to click into person edit and see the old consents. I removed them from there. They are viewable on the consents tab on this page:
Screen Shot 2021-05-27 at 11 34 12 AM

**Review by pbanaszkiewicz — APPROVED** [review]

**Pull request merged**